lib/tracker.rb

```diff
@@ -0,0 +1,26 @@
+require_relative 'transaction'
+require 'date'
+
+class Tracker
+  attr_reader :balance
+
+  def initialize
+    @balance = 0
+  end
+
+  def record_debit(amount, transaction)
```

Comment:
The injection of a transaction object really helps with decoupling.  What is the downside of having a transaction that initially has no value set on it (when it's created and injected) and then has its correct value set when `set_debit` is called? Is there a way to inject Transaction that avoids this downside?